spec/support/shared_examples/can_present_entity_display_name.rb

```diff
@@ -1,101 +1,113 @@
 # frozen_string_literal: true
 
-RSpec.shared_examples "Can present entity display name" do
+RSpec.shared_examples "Can present entity display name" do |factory:|
 
-  include_context("Sample registration with defaults")
+  include_context "Sample registration with defaults", factory
+  let(:registration) { resource }
 
-  subject { described_class.new(registration, view) }
+  let(:upper_tier) { WasteCarriersEngine::Registration::UPPER_TIER }
+  let(:lower_tier) { WasteCarriersEngine::Registration::LOWER_TIER }
 
-  shared_examples "trading as" do
+  subject { registration.entity_display_name }
+
+  shared_examples "legal_entity_name trading as trading_as_name" do
+    it "returns entity name trading as business name with truncation" do
+      expect(subject).to eq "#{registration.legal_entity_name} trading as #{trading_as_name}"
+    end
+  end
+
+  shared_examples "legal_entity_name trading as company_name" do
     it "returns entity name trading as business name" do
-      expect(subject.entity_display_name).to eq "#{entity_name} trading as #{trading_as_name}"
+      expect(subject).to eq "#{registration.legal_entity_name} trading as #{company_name}"
     end
   end
 
-  shared_examples "with and without a business name" do
-    context "without a business name" do
-      let(:company_name) { nil }
+  shared_examples "simply company_name" do
+    it "returns business name" do
+      expect(subject).to eq company_name
+    end
+  end
 
-      it "returns the entity name" do
-        expect(subject.entity_display_name).to eq entity_name
-      end
+  shared_examples "simply legal_entity_name" do
+    it "returns the legal entity name" do
+      expect(subject).to eq registration.legal_entity_name
     end
+  end
 
-    context "with a business name" do
-      let(:trading_as_name) { Faker::Company.name }
+  shared_examples "with a business name with truncation" do
```

Comment:
Can we clarify what a 'business name'  is in this instance? I think it's a `company_name` ?